packages/terra-table/src/Table.jsx

```diff
@@ -1,68 +1,401 @@
+      >
+        {createHeaderCheckCell(headerData.selectAllColumn, rowStyle, checkStyle)}
+        {headerData.cells.map((cellData, colIndex) => (
+          <HeaderCell
+            {...cellData.attrs}
+            key={cellData.key}
+            refCallback={cellData.refCallback}
+            metaData={cellData.metaData}
+            isSortDesc={cellData.isSortDesc}
+            isSortActive={cellData.isSortActive}
+            onCellAction={cellData.onCellAction}
+            onSortAction={cellData.onSortAction}
+            removeInner={cellData.removeInner}
+            width={columnWidths ? columnWidths[colIndex] : undefined}
+          >
+            {cellData.children}
+          </HeaderCell>
+        ))}
+        {createHeaderChevronCell(rowStyle, hasChevrons)}
+      </HeaderRow>
+    ),
+  };
+};
+
+const unpackTableData = (headerData, sectionData, columnWidths, rowStyle, checkStyle, hasChevrons, dividerStyle) => {
+  const { headerIndex, header } = createHeader(headerData, columnWidths, rowStyle, checkStyle, hasChevrons);
+  const { sectionIndex, sections } = createSections(headerIndex, sectionData, headerData, columnWidths, rowStyle, checkStyle, hasChevrons, dividerStyle);
+  return { rowCount: sectionIndex, header, sections };
 };
 
 const Table = ({
-  children,
-  isStriped,
-  isPadded,
+  dividerStyle,
+  hasChevrons,
+  rowStyle,
+  checkStyle,
+  columnWidths,
+  headerData,
+  sectionData,
+  fill,
+  footerNode,
+  headerNode,
+  paddingStyle,
+  numberOfRows,
+  scrollRefCallback,
+  summary,
+  summaryId,
   ...customProps
 }) => {
-  const tableClassNames = cx([
+  const attrSpread = {};
+  if (paddingStyle !== 'none') {
+    attrSpread['data-table-padding'] = paddingStyle;
+  }
+
+  const makeInline = columnWidths && columnWidths.length ? columnWidths.every(width => !!width.static) : undefined;
+  const hasEndNodes = headerNode || footerNode;
+
+  const tableClasses = cx(
     'table',
-    { striped: isStriped },
-    { padded: isPadded },
-    customProps.className,
-  ]);
+    { fill },
+    { 'is-inline': makeInline },
+    { rounded: !hasEndNodes },
+  );
+
+  const { rowCount, header, sections } = unpackTableData(headerData, sectionData, columnWidths, rowStyle, checkStyle, hasChevrons, dividerStyle);
+
+  const rows = (
+    <div
+      {...customProps}
+      {...attrSpread}
+      className={customProps.className ? `${tableClasses} ${customProps.className}` : tableClasses}
+      role="grid"
+      aria-rowcount={numberOfRows || rowCount}
+      aria-describedby={summaryId}
+    >
+      <VisuallyHiddenText id={summaryId} text={summary} />
+      {header}
+      {sections ? (
+        <div className={cx(['body'])} role="rowgroup" ref={scrollRefCallback}>
+          {sections}
+        </div>
+      ) : undefined}
+    </div>
+  );
+
+  if (!hasEndNodes) {
+    return rows;
+  }
+
   return (
-    <table {...customProps} className={tableClassNames}>
-      {children}
-    </table>
+    <ContentContainer
```

Comment:
Why are we using the `<ContentContainer>` instead of rendering the rows standalone? If consumers wanted headers/footers, they could drop their table into a `ContentContainer` directly?

Reply:
There is a specific border style that UX wants for this to make them seemed grouped, so it's built in.